app/src/main/java/space/taran/arknavigator/mvp/model/UserPreferences.kt

```diff
@@ -39,8 +48,38 @@ class UserPreferences @Inject constructor(val context: Context) {
         else Sorting.values()[intValue]
     }
 
+    suspend fun setCrashReportEnabled(enabled: Boolean?) {
+        dataStore.edit { preferences ->
+            preferences[PreferencesKeys.CRASH_REPORT_PREFERENCE] = enabled?: BuildConfig.DEBUG
+        }
+    }
+
+    suspend fun isCrashReportEnabled(): Boolean =
+        dataStore.data.first()[PreferencesKeys.CRASH_REPORT_PREFERENCE]?: BuildConfig.DEBUG
+
+    suspend fun setCacheReplicationEnabled(enabled: Boolean) {
+        dataStore.edit { preferences ->
+            preferences[PreferencesKeys.IMG_CACHE_REPLICATION_PREF] = enabled
+        }
+    }
+
+    suspend fun isCacheReplicationEnabled(): Boolean =
+        dataStore.data.first()[PreferencesKeys.IMG_CACHE_REPLICATION_PREF] ?: true
```

Comment:
Do we really need to duplicate default values in `clearPreferences` and `isSomethingEnabled` methods?
If yes, let's make constants and use them to avoid inconsistency.

Reply:
> Do we really need to duplicate default values in `clearPreferences` and `isSomethingEnabled` methods? If yes, let's make constants and use them to avoid inconsistency.

Yes, because we should take care of that one scenario on a freshly installed app. This scenario also extends whenever the user clears the caches of the app.